### ISSUE TITLE
5.1.2 (#560) breaks sanitization in OneOf

### ISSUE THREAD

**Issue by hijohnnylin** [express-validator/express-validator#565]
UPDATE 4/26:

Sanitization seems to be broken in 5.1.2 when inside a OneOf. It works correctly in 5.1.1. See example below.

Code
```
router.get("/sanitization-test",
[
  oneOf(
    [
      [
        check("TrimMe1")
          .trim()
      ],
      [
        check("TrimMe2")
          .trim()
      ]
    ]
  )
],
(request, response, next) => {
  const result = validationResult(request)
  response.status(200).json({
    TrimMe1: matchedData(request).TrimMe1
  });
});
```

Request
`http://localhost:3000/sanitization-test?TrimMe=%20%20ShouldBeTrimmed%20%20`

Expected Response (5.1.1):
`{"TrimMe1":"ShouldBeTrimmed"}`

Actual Response (5.1.2):
`{"TrimMe1":"  ShouldBeTrimmed  "}` 

==== OUTDATED POST BELOW PRESERVED FOR RECORD ===

Tested to work correctly in 5.1.1, but 5.1.2 (due to #560) behaves very differently. Issues:
1) double sanitization leads to silent failure of the second sanitization: `.trim().normalizeEmail()` leads to `normalizeEmail()` being ignored
2) double sanitization of field A leads to failure of all sanitization in field B

Recommended Fix:
For 1), this may be a case of "behaves correctly", but I think instead of just silently failing the second sanitization (normalizeEmail in this case), it should throw an error.
For 2), this should be fixed. Ignoring double-sanitization on one field should not completely stop all sanitization on other fields.

Code:
```
check("email")
          .exists().withMessage("Missing email address.")
          .isEmail().withMessage("Invalid email address.")
          .trim(),
          .normalizeEmail(),
check("someBooleanString")
          .toBoolean()
```

Input:
```
email = "yo+blah@gmail.com"
someBooleanString = "true"
```

Expected MatchedData Output (correct in 5.1.1):
```
email = "yo@gmail.com"
someBooleanString = true (Boolean, not String)
```

5.1.2 Output:
```
email = "yo+blah@gmail.com"
someBooleanString = "true" (String, not sanitized)
```

**Comment by gustavohenke** [comment]
Hey @hijohnnylin! Thanks for the report. Sounds like a serious issue.
I'm short on time lately, so if you could take a stab at this, I would be thankful!

**Comment by gustavohenke** [comment]
Hi @hijohnnylin, I tried your snippet, and it works.
Are you sure that's all? Can you provide a reproducible example?

**Comment by hijohnnylin** [comment]
Hi there,
Sorry about the delay in checking up on this. I realized that the issue only happens when it's in a OneOf. I've updated the example code to reproduce the issue below and also updated the original post. Seems that sanitization isn't working at all in a OneOf, so I've simplified the example below.

Code
```
router.get("/sanitization-test",
[
  oneOf(
    [
      [
        check("TrimMe1")
          .trim()
      ],
      [
        check("TrimMe2")
          .trim()
      ]
    ]
  )
],
(request, response, next) => {
  const result = validationResult(request)
  response.status(200).json({
    TrimMe1: matchedData(request).TrimMe1
  });
});
```

Request
`http://localhost:3000/sanitization-test?TrimMe=%20%20ShouldBeTrimmed%20%20`

Expected Response (5.1.1):
`{"TrimMe1":"ShouldBeTrimmed"}`

Actual Response (5.1.2):
`{"TrimMe1":"  ShouldBeTrimmed  "}`
